_posts/2019-02-01-update-a-secondary-ecu.adoc

```diff
@@ -0,0 +1,109 @@
+= Update a secondary ECU
+:page-layout: page
+:page-categories: [using-libaktualizr]
+:page-date: 2018-11-28 14:06:25
+:page-order: 6
+:icons: font
+
+This page shows an example of how to use libaktualizr API to update
+a secondary ECU.
+
+[source,cpp]
+----
+boost::filesystem::path cfg_path("/var/sota/sota_local.toml");
+Aktualizr aktualizr(cfg_path);
+----
+This will create a new aktualizr instance based on the provided config file.
+Next, configure a secondary and add it to aktualizr:
+[source,cpp]
+----
+Uptane::SecondaryConfig sconfig;
+sconfig.secondary_type = Uptane::SecondaryType::kVirtual;
+sconfig.ecu_serial = "9b6abd606a761074df0092606465ddc9";
+sconfig.ecu_hardware_id = "TCU";
+sconfig.full_client_dir = "/var/sota/ecus/tcu";
+sconfig.ecu_private_key = "sec.private";
+sconfig.ecu_public_key = "sec.public";
+sconfig.target_name_path = "target_name";
+sconfig.metadata_path = "/var/sota/ecus/tcu/metadata";
+auto secondary = std::make_shared<Uptane::VirtualSecondary>(sconfig);
+aktualizr.AddSecondary(secondary);
+----
+
+Initialize call will provision the device on the backend:
+[source,cpp]
+----
+aktualizr.Initialize();
+----
+
+Then provide the device information (like list of ECUs,
+installed software versions, etc.) to the backend.
+[source,cpp]
+----
+aktualizr.SendDeviceData().get();
+----
+
+Now it is possible to query the server about currently running campaigns for
+the device:
+[source,cpp]
+----
+auto result = aktualizr.CampaignCheck().get();
+if (result.campaigns.empty()) {
+  std::cout << "System is up to date" <<std::endl;
+}
+----
+
+If there are some running campaigns, a vehicle owner must first give his consent
```

Comment:
"his" can be removed. It solves the gendered language problem and is just as clear without it.